hangman.py

```diff
@@ -0,0 +1,427 @@
+                      get_guessed_word(secret_word, letters_guessed))
+                continue
+            else:
+                warnings_left -= 1
+                print("Oops! That is not a valid letter. You have", warnings_left, "warnings left:",
+                      get_guessed_word(secret_word, letters_guessed))
+                continue
+
+        if guessed_letter in letters_guessed:
+            if warnings_left == 0:
+                guesses_left -= 1
+                print("Oops! You've already guessed that letter. You have no warnings left so you lose one guess:",
+                      get_guessed_word(secret_word, letters_guessed))
+                continue
+            else:
+                warnings_left -= 1
+                print("Oops! You've already guessed that letter. You have", warnings_left, "warnings left:",
+                      get_guessed_word(secret_word, letters_guessed))
+                continue
+        else:
+            letters_guessed.append(guessed_letter)
+
+        if guessed_letter in letters_in_secret_word:
+            print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+        else:
+            if guessed_letter in list_of_en_golos:
+                guesses_left -= 2
+            else:
+                guesses_left -= 1
+            print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+
+        if is_word_guessed(secret_word, letters_guessed):
+            total_score = guesses_left * score(secret_word, letters_guessed)
+            print("Congratulations, you won! Your total score for this game is:", total_score)
+            break
+
+
+
+# When you've completed your hangman function, scroll down to the bottom
+# of the file and uncomment the first two lines to test
+#(hint: you might want to pick your own
+# secret_word while you're doing your own testing)
+
+
+# -----------------------------------
+
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    my_word_list = []
+    other_word_list = []
+
+    for x in my_word:
+        if x == ' ':
+            continue
+        my_word_list.append(x)
+
+    # Потрібно прописати випадок для 'a_ ple' 'apple'
+    for x in other_word:
+        other_word_list.append(x)
+
+    if len(my_word_list) != len(other_word_list):
+        return False
+
+    for x in range(0, len(my_word_list)):
+        if my_word_list[x] == '_' or my_word_list[x] == other_word_list[x]:
+            continue
+        else:
+            return False
+
+    my_word_list_without_gaps = []
```

Comment:
It's hard to understand the rest of code here. Wither try to refactor it, or add comments